manifests/option.pp

```diff
@@ -13,7 +13,7 @@
 
 define samba::option (
   String $key         = $title,
-  Variant[Boolean, Integer, String, Array[String], Undef] $value = undef,
+  Variant[Hash, Boolean, Integer, String, Array[String], Undef] $value = undef,
```

Comment:
I think the samba:option is not the right place to add a hash type. It will be very complex to pass it on to the `augeas`. I think this needs to be handled in a different way. I am not fully sure yet but it looks very similar to the this module handles the shares.

Reply:
Indeed, I have been having trouble with Hash and augeas.  I'll be working on the code more today and tomorrow; perhaps I can come up with a solution.

Perhaps we could abandon augeas altogether?  I've always thought of augeas as the last choice when managing configuration files.  Yes, it would be significant effort in moving to template-based management of the smb.conf file but we'd have significantly more control over the creation and validation of the smb.conf parameters.

Reply:
The reason I prefer augeas over templates is the constant upkeep required to align the template with upstream changes. It is just not worth the effort across many different platforms. This hash can be solved similar to how shares work. They are simply hash values as well.